shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

```diff
@@ -0,0 +1,143 @@
+    _handlers = [[NSMutableDictionary<NSString*, FlutterBinaryMessageHandler> alloc] init];
+  }
+  return self;
+}
+
+- (void)sendOnChannel:(NSString*)channel message:(NSData* _Nullable)message {
+  [self sendOnChannel:channel message:message binaryReply:nil];
+}
+
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  self.channel = channel;
+  self.message = message;
+}
+
+- (void)setMessageHandlerOnChannel:(NSString*)channel
+              binaryMessageHandler:(FlutterBinaryMessageHandler _Nullable)handler {
+  [self.handlers setObject:handler forKey:channel];
+}
+@end
+
+@interface FlutterChannelsTest : XCTestCase
+@end
+
+@implementation FlutterChannelsTest
+
+- (void)testMethodInvoke {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  [channel invokeMethod:@"foo" arguments:@[ @(1) ]];
+  OCMVerify([binaryMessenger sendOnChannel:channelName message:encodedMethodCall]);
+}
+
+- (void)testMethodInvokeWithReply {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  __block BOOL didCallReply = NO;
+  OCMExpect([binaryMessenger sendOnChannel:channelName
+                                   message:encodedMethodCall
+                               binaryReply:[OCMArg checkWithBlock:^BOOL(id obj) {
+                                 FlutterBinaryReply reply = obj;
+                                 reply(nil);
+                                 return YES;
+                               }]]);
+  [channel invokeMethod:@"foo"
+              arguments:@[ @(1) ]
+                 result:^(id _Nullable result) {
+                   didCallReply = YES;
+                   XCTAssertEqual(FlutterMethodNotImplemented, result);
+                 }];
+  OCMVerify(binaryMessenger);
+  XCTAssertTrue(didCallReply);
```

Comment:
I'm not familiar with this code, is `invokeMethod:::` async?  Do you need to `- waitForExpectations:timeout:` before the verifies and asserts?

Reply:
In practice it's usually async.  By using checkWithBlock and calling the callback I've made it synchronous ie there are no dispatches in the code anywhere.